helix-term/src/ui/picker.rs

```diff
@@ -61,11 +71,33 @@ impl<T> FilePicker<T> {
     }
 
     fn calculate_preview(&mut self, editor: &Editor) {
```

Comment:
If we change `calculate_preview` to this:

```rust
fn get_preview(&mut self, path: &Path, editor: &Editor) -> Preview {/* ... */}
```

We can then do `let cache = self.get_preview(&path, &cx.editor);` in L152 and remove the `self.calculate_preview(cx.editor);` call at the beginning of `render()`, which will also  prevent unnecessary preview calculation when the window is too small.

Reply:
What do you think implement constructor on Preview like this?

```rust
enum Preview {
    Document(Document),
    Binary,
    LargeFile,
    NotFound,
}

impl Preveiw {
     fn new(
        path: impl AsRef<Path>,
        read_buffer: &mut Vec<u8>,
        /* argument same as Document::open */
    )->Preview{
        ..
    }
}
```

Reply:
I think it's better to just construct the `Preview` directly inside `get_preview`